## Animals Classification

Numpy:- In this Project we are using 'Numpy' Library for Numbers Operation here.

Pandas:- In this Project we are using 'Pandas' Library for EDA and data related task.

Matplotlib:- 'Matplotlib' we are using for Visualization Purpose.

#### Importing Libraries

In [1]:
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt

#### Importing Data

In [2]:
df = pd.read_csv('Animals Classification/zoo.csv')

df.head()

animal_name  hair  feathers  eggs  milk  airborne  aquatic  predator  \
0    aardvark     1         0     0     1         0        0         1   
1    antelope     1         0     0     1         0        0         0   
2        bass     0         0     1     0         0        1         1   
3        bear     1         0     0     1         0        0         1   
4        boar     1         0     0     1         0        0         1   

   toothed  backbone  breathes  venomous  fins  legs  tail  domestic  catsize  \
0        1         1         1         0     0     4     0         0        1   
1        1         1         1         0     0     4     1         0        1   
2        1         1         0         0     1     0     1         0        0   
3        1         1         1         0     0     4     0         0        1   
4        1         1         1         0     0     4     1         0        1   

   class_type  
0           1  
1           1  
2           4  
3           1  
4           1

In [3]:
features = list(df.columns)
print(features)

['animal_name', 'hair', 'feathers', 'eggs', 'milk', 'airborne', 'aquatic', 'predator', 'toothed', 'backbone', 'breathes', 'venomous', 'fins', 'legs', 'tail', 'domestic', 'catsize', 'class_type']


In [4]:
features.remove('class_type')
features.remove('animal_name')

print(features)

['hair', 'feathers', 'eggs', 'milk', 'airborne', 'aquatic', 'predator', 'toothed', 'backbone', 'breathes', 'venomous', 'fins', 'legs', 'tail', 'domestic', 'catsize']


In [5]:
X = df[features].values.astype(np.float32)
Y = df.class_type


print(X.shape)
print(Y.shape)

(101, 16)
(101,)


In [7]:
from sklearn.model_selection import train_test_split

X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 0.5, random_state = 0)

print(X_train.shape)
print(Y_train.shape)
print(X_test.shape)
print(Y_test.shape)

(50, 16)
(50,)
(51, 16)
(51,)


### Using Logistic Regression

In [8]:
from sklearn.linear_model import LogisticRegression

model = LogisticRegression()

In [9]:
model.fit(X_train, Y_train)
print("training accuracy :", model.score(X_train, Y_train))
print("testing accuracy :", model.score(X_test, Y_test))

training accuracy : 1.0
testing accuracy : 0.9215686274509803


### Using Decision Tree Classifier

In [10]:
from sklearn.tree import DecisionTreeClassifier

model = DecisionTreeClassifier()

In [11]:
model.fit(X_train, Y_train)
print("training accuracy :", model.score(X_train, Y_train))
print("testing accuracy :", model.score(X_test, Y_test))

training accuracy : 1.0
testing accuracy : 0.9019607843137255


### Using Random Forest

In [12]:
from sklearn.ensemble import RandomForestClassifier

model = RandomForestClassifier()

In [13]:
model.fit(X_train, Y_train)
print("training accuracy :", model.score(X_train, Y_train))
print("testing accuracy :", model.score(X_test, Y_test))

training accuracy : 1.0
testing accuracy : 0.9215686274509803


### Using Support Vector Machine

In [14]:
from sklearn.svm import SVC

model = SVC(kernel = "rbf", C = 1.0, gamma = 0.1)

In [15]:
model.fit(X_train, Y_train)
print("training accuracy :", model.score(X_train, Y_train))
print("testing accuracy :", model.score(X_test, Y_test))

training accuracy : 0.98
testing accuracy : 0.8235294117647058


### Using Ada Boost Classifier

In [16]:
from sklearn.ensemble import AdaBoostClassifier

model = AdaBoostClassifier()

In [17]:
model.fit(X_train, Y_train)
print("training accuracy :", model.score(X_train, Y_train))
print("testing accuracy :", model.score(X_test, Y_test))

training accuracy : 0.82
testing accuracy : 0.5882352941176471


#### As we have use few algorithm here. So you can see the better accuracy level according to algorithm below:

LogisticRegression :- As you have seen above in 'LogisticRegression' it's providing accuracy level = 92%.

DecisionTreeClassifier :- As you have seen above in 'DecisionTreeClassifier' it's providing accuracy level = 90%.

RandomForestRegressor :- As you have seen above in 'RandomForestRegressor' it's providing accuracy level = 92%

SupportVectorMachine :- As you have seen above in 'SupportVectorMachine' it's providing accuracy level = 82%

AdaBoostClassifierr :- As you have seen above in 'AdaBoostClassifier' it's providing accuracy level = 59%
